modin/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -0,0 +1,1270 @@
+            )
+        )
+        return _register_default_pandas(
+            _register_expanding(pandas.core.window.expanding.Expanding.corr)
+        )(
+            self,
+            fold_axis,
+            expanding_args,
+            other=other_for_default,
+            pairwise=pairwise,
+            ddof=ddof,
+            numeric_only=numeric_only,
+            squeeze_self=squeeze_self,
+            **kwargs,
+        )
+
+    def groupby_size(
+        self,
+        by,
+        axis,
+        groupby_kwargs,
+        agg_args,
+        agg_kwargs,
+        drop=False,
+    ):
+        result = _register_default_pandas(_groupby("size"))(
+            self,
+            by=by,
+            axis=axis,
+            groupby_kwargs=groupby_kwargs,
+            agg_args=agg_args,
+            agg_kwargs=agg_kwargs,
+            drop=drop,
+            method="size",
+        )
+        if not groupby_kwargs.get("as_index", False):
+            # Renaming 'MODIN_UNNAMED_SERIES_LABEL' to a proper name
+
+            result.columns = result.columns[:-1].append(pandas.Index(["size"]))
+        return result
+
+    def get_axis(self, axis):
+        return self._modin_frame.index if axis == 0 else self._modin_frame.columns
+
+    def get_index_name(self, axis=0):
+        return self.get_axis(axis).name
+
+    def get_index_names(self, axis=0):
+        return self.get_axis(axis).names
+
+    def set_index_name(self, name, axis=0):
+        self.get_axis(axis).name = name
+
+    def has_multiindex(self, axis=0):
+        if axis == 0:
+            return isinstance(self._modin_frame.index, pandas.MultiIndex)
+        assert axis == 1
+        return isinstance(self._modin_frame.columns, pandas.MultiIndex)
+
+    def isin(self, values, ignore_indices=False, **kwargs):
+        if isinstance(values, type(self)) and ignore_indices:
+            # Pandas logic is that it ignores indexing if 'values' is a 1D object
+            values = values.to_pandas().squeeze(axis=1)
+        if self._shape_hint == "column":
+            return _register_default_pandas(pandas.Series.isin, is_series=True)(
+                self, values, **kwargs
+            )
+        else:
+            return _register_default_pandas(pandas.DataFrame.isin)(
+                self, values, **kwargs
+            )
+
+    def to_pandas(self):
+        return self._modin_frame
+
+    @classmethod
+    def from_pandas(cls, df, data_cls):
+        return cls(df)
+
+    @classmethod
+    def from_arrow(cls, at, data_cls):
+        return
+
+    def free(self):
+        return
+
+    def finalize(self):
+        return
+
+    # Dataframe exchange protocol
+
+    def to_dataframe(self, nan_as_null: bool = False, allow_copy: bool = True):
+        return self._modin_frame.__dataframe__(
+            nan_as_null=nan_as_null, allow_copy=allow_copy
+        )
+
+    @classmethod
+    def from_dataframe(cls, df, data_cls):
+        return cls(data_cls.from_dataframe(df))
```

Comment:
Can you check if pandas df has `from_dataframe` and `__dataframe__` methods?

Reply:
pandas df does not support from_dataframe, change to pandas.api.interchange.from_dataframe